backend/src/sockets/handler.ts

```diff
@@ -5,64 +5,87 @@ import logger from "../config/logger";
-    for (const [event, handler] of Object.entries(handlers)) {
-      if (handler) {
-        socket.on(event, async data => {
-          await handler(io, socket, roomId, data);
-        });
+  constructor(database: Database, io: Server, socket: Socket, roomId: string) {
+    this.database = database;
+    this.io = io;
+    this.socket = socket;
+    this.roomId = roomId;
+  }
+
+  initialize(): void {
+    this.socket.join(this.roomId, () => {
+      const handlers = this.createEventHandler();
+
+      this.socket.to(this.roomId).emit(Event.MESSAGE, "A new user has joined the room!");
+      for (const [event, handler] of Object.entries(handlers)) {
+        if (handler) {
+          this.socket.on(event, async data => {
+            await handler(data);
+          });
+        }
       }
-    }
-  });
-};
+    });
+  }
 
-handlers[Event.PLAY_VIDEO] = (io: Server, socket: Socket, roomId: string, time: number): Promise<void> => {
-  socket.to(roomId).emit(Event.PLAY_VIDEO, time);
-  return Promise.resolve();
-};
+  private createEventHandler(): EventHandler {
+    return {
+      [Event.PLAY_VIDEO]: (time: number): Promise<void> => {
+        this.socket.to(this.roomId).emit(Event.PLAY_VIDEO, time);
+        return Promise.resolve();
+      },
 
-handlers[Event.PAUSE_VIDEO] = (io: Server, socket: Socket, roomId: string, time: number): Promise<void> => {
-  socket.to(roomId).emit(Event.PAUSE_VIDEO, time);
-  return Promise.resolve();
-};
+      [Event.PAUSE_VIDEO]: (time: number): Promise<void> => {
+        this.socket.to(this.roomId).emit(Event.PAUSE_VIDEO, time);
+        return Promise.resolve();
+      },
 
-handlers[Event.REQUEST_ADD_TO_QUEUE] = async (
-  io: Server,
-  socket: Socket,
-  roomId: string,
-  videoUrl: string
-): Promise<void> => {
-  logger.info("REQUEST_ADD_TO_QUEUE called!");
-
-  try {
-    // TODO: Improve method of finding video ID to account for extra query parameters
-    const videoId = videoUrl.split("v=")[1];
-
-    const videoInfoResponse = await axios.get(`https://youtube.com/get_video_info?video_id=${videoId}`);
-    const videoInfo = qs.parse(videoInfoResponse.data);
-    const playerResponse = JSON.parse(videoInfo["player_response"] as string);
-    const videoTitle = playerResponse.videoDetails.title;
-    logger.info(`Video found: ${videoTitle}`);
-
-    const video: Video = { id: uniqid(), title: videoTitle, url: videoUrl };
-    io.in(roomId).emit(Event.ADD_TO_QUEUE, video);
-  } catch {
-    logger.error("Failed to find info about video");
-  }
-};
+      [Event.REQUEST_ADD_TO_QUEUE]: async (videoUrl: string): Promise<void> => {
+        logger.info("REQUEST_ADD_TO_QUEUE called!");
 
-handlers[Event.SET_VIDEO] = (io: Server, socket: Socket, roomId: string, video: Video): Promise<void> => {
-  io.in(roomId).emit(Event.SET_VIDEO, video);
-  return Promise.resolve();
-};
+        try {
+          // TODO: Improve method of finding video ID to account for extra query parameters
```

Comment:
We should pull out the code that does the video validation check into its own function. Maybe it can go into a separate util file for now? I also think it would be more efficient if we did the validation on the client side as it would reduce an unnecessary request to our server. I think @kriscupofty is working on the video validation as well. You two should talk to each other so you're not both working on the same thing.